trio/_signals.py

```diff
@@ -144,6 +144,9 @@ def open_signal_receiver(*signals):
                  reload_configuration()
 
     """
+    if not signals:
+        raise AttributeError("No signals were provided")
```

Comment:
AttributeError is reserved to attributes of objects, I think ValueError is more appropriate here.

Reply:
Or maybe `TypeError`? That's what you get if you try to call a function with too few mandatory arguments:

```
In [1]: def f(x): 
   ...:     pass 
   ...:                                                                                   

In [2]: f()                                                                               
TypeError: f() missing 1 required positional argument: 'x'
```

We should also add a newsfragment.

Reply:
What's that 'newsfragment' ?

Reply:
changed to `TypeError`

Reply:
See https://trio.readthedocs.io/en/stable/contributing.html#pull-request-release-notes for details